Can't delete new resource

#279 is not fixed completely, I can delete old resources but new ones can't be delete:
1. Create a new resource, doesn't matter if it is an app, appServer or other resource: aaaa
2. Delete resource aaaa via resource list screen:
Error on GUI: `Could not remove resourcetype.`

Error in Log:
```
2017-12-13 17:45:13,924 logLev=WARN class=org.hibernate.hql.internal.ast.QueryTranslatorImpl thread="ajp-/0.0.0.0:8009-6" HHH000104: firstResult/maxResults specified with collection fetch; applying in memory!
2017-12-13 17:45:14,915 logLev=INFO class=ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary thread="ajp-/0.0.0.0:8009-6" Resource with id: 256360 is going to be removed from the database...
2017-12-13 17:45:15,374 logLev=INFO class=ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceRepository thread="ajp-/0.0.0.0:8009-6" Resource with Id: 256360 was removed from the db
2017-12-13 17:45:15,402 logLev=WARN class=org.hibernate.engine.jdbc.spi.SqlExceptionHelper thread="ajp-/0.0.0.0:8009-6" SQL Error: 2292, SQLState: 23000
2017-12-13 17:45:15,403 logLev=ERROR class=org.hibernate.engine.jdbc.spi.SqlExceptionHelper thread="ajp-/0.0.0.0:8009-6" ORA-02292: integrity constraint (BTO_YAMW.FK_RESTRICTION_04) violated - child record found

2017-12-13 17:45:15,407 logLev=ERROR class=org.jboss.as.ejb3.invocation thread="ajp-/0.0.0.0:8009-6" JBAS014134: EJB Invocation failed on component ResourceBoundary for method public void ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary.removeResourceEntityOfDefaultResType(ch.puzzle.itc.mobiliar.business.foreignable.entity.ForeignableOwner,java.lang.Integer) throws ch.puzzle.itc.mobiliar.common.exception.ResourceNotFoundException,ch.puzzle.itc.mobiliar.common.exception.ElementAlreadyExistsException,ch.puzzle.itc.mobiliar.business.foreignable.entity.ForeignableOwnerViolationException: javax.ejb.EJBException: javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.handleExceptionInOurTx(CMTTxInterceptor.java:191)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:281)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.requiresNew(CMTTxInterceptor.java:353)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:245)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processInvocation(LoggingInterceptor.java:59)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.AdditionalSetupInterceptor.processInvocation(AdditionalSetupInterceptor.java:55)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.TCCLInterceptor.processInvocation(TCCLInterceptor.java:45)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ViewService$View.invoke(ViewService.java:189)
        at org.jboss.as.ee.component.ViewDescription$1.processInvocation(ViewDescription.java:185)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:73)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary$$$view17.removeResourceEntityOfDefaultResType(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor1027.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.weld.util.reflection.SecureReflections$13.work(SecureReflections.java:267)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.run(SecureReflectionAccess.java:52)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.runAsInvocation(SecureReflectionAccess.java:137)
        at org.jboss.weld.util.reflection.SecureReflections.invoke(SecureReflections.java:263)
        at org.jboss.weld.bean.proxy.EnterpriseBeanProxyMethodHandler.invoke(EnterpriseBeanProxyMethodHandler.java:115)
        at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
        at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:105)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary$Proxy$_$$_Weld$Proxy$.removeResourceEntityOfDefaultResType(ResourceBoundary$Proxy$_$$_Weld$Proxy$.java)
        at ch.puzzle.itc.mobiliar.presentation.resourcelist.ResourceListController.removeResource(ResourceListController.java:91)
        at ch.puzzle.itc.mobiliar.presentation.resourcelist.ResourceListController$Proxy$_$$_WeldClientProxy.removeResource(ResourceListController$Proxy$_$$_WeldClientProxy.java)
        at ch.puzzle.itc.mobiliar.presentation.resourcelist.ResourceListDataProvider.deleteResourceAction(ResourceListDataProvider.java:216)
        at ch.puzzle.itc.mobiliar.presentation.resourcelist.ResourceListDataProvider.deleteResource(ResourceListDataProvider.java:193)
        at ch.puzzle.itc.mobiliar.presentation.resourcelist.ResourceListDataProvider$Proxy$_$$_WeldClientProxy.deleteResource(ResourceListDataProvider$Proxy$_$$_WeldClientProxy.java)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_144]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.apache.el.parser.AstValue.invoke(AstValue.java:258)
        at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:278)
        at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40)
        at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50)
        at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:105)
        at javax.faces.component.MethodBindingMethodExpressionAdapter.invoke(MethodBindingMethodExpressionAdapter.java:87)
        at com.sun.faces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:101)
        at javax.faces.component.UICommand.broadcast(UICommand.java:315)
        at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:786)
        at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:1251)
        at com.sun.faces.lifecycle.InvokeApplicationPhase.execute(InvokeApplicationPhase.java:81)
        at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
        at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:118)
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:593)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:295)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
        at ch.puzzle.itc.mobiliar.presentation.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:43)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:246)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
        at ch.puzzle.itc.mobiliar.presentation.util.CachingFilter.doFilter(CachingFilter.java:65)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:246)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:231)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:149)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.event(JBossWebContext.java:91)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.invoke(JBossWebContext.java:72)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:512)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.web.security.SecurityContextAssociationValve.invoke(SecurityContextAssociationValve.java:169)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:151)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:97)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:102)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:490)
        at org.apache.coyote.ajp.AjpProtocol$AjpConnectionHandler.process(AjpProtocol.java:422)
        at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:926)
        at java.lang.Thread.run(Thread.java:748) [rt.jar:1.8.0_144]
Caused by: javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1387)
        at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1310)
        at org.hibernate.ejb.AbstractEntityManagerImpl.throwPersistenceException(AbstractEntityManagerImpl.java:1397)
        at org.hibernate.ejb.AbstractQueryImpl.executeUpdate(AbstractQueryImpl.java:111)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceGroupRepository.preserveDeployments(ResourceGroupRepository.java:198)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceGroupRepository.remove(ResourceGroupRepository.java:191)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary.doRemoveResourceEntity(ResourceBoundary.java:356)
        at ch.puzzle.itc.mobiliar.business.resourcegroup.boundary.ResourceBoundary.removeResourceEntityOfDefaultResType(ResourceBoundary.java:306)
        at sun.reflect.GeneratedMethodAccessor1027.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.as.ee.component.ManagedReferenceMethodInterceptor.processInvocation(ManagedReferenceMethodInterceptor.java:52)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:375)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.doMethodInterception(Jsr299BindingsInterceptor.java:86)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.processInvocation(Jsr299BindingsInterceptor.java:97)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.ExecutionTimeInterceptor.processInvocation(ExecutionTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.jpa.interceptor.SBInvocationInterceptor.processInvocation(SBInvocationInterceptor.java:47)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:73)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InitialInterceptor.processInvocation(InitialInterceptor.java:21)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.interceptors.ComponentDispatcherInterceptor.processInvocation(ComponentDispatcherInterceptor.java:53)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.pool.PooledInstanceInterceptor.processInvocation(PooledInstanceInterceptor.java:51)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:279)
        ... 84 more
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:74)
        at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:49)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:124)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:109)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:189)
        at org.hibernate.persister.entity.AbstractEntityPersister.delete(AbstractEntityPersister.java:3358)
        at org.hibernate.persister.entity.AbstractEntityPersister.delete(AbstractEntityPersister.java:3558)
        at org.hibernate.action.internal.EntityDeleteAction.execute(EntityDeleteAction.java:102)
        at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:395)
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:387)
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:310)
        at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:349)
        at org.hibernate.event.internal.DefaultAutoFlushEventListener.onAutoFlush(DefaultAutoFlushEventListener.java:67)
        at org.hibernate.internal.SessionImpl.autoFlushIfRequired(SessionImpl.java:1166)
        at org.hibernate.internal.SessionImpl.executeUpdate(SessionImpl.java:1246)
        at org.hibernate.internal.QueryImpl.executeUpdate(QueryImpl.java:116)
        at org.hibernate.ejb.QueryImpl.internalExecuteUpdate(QueryImpl.java:198)
        at org.hibernate.ejb.AbstractQueryImpl.executeUpdate(AbstractQueryImpl.java:102)
        ... 118 more
Caused by: java.sql.SQLIntegrityConstraintViolationException: ORA-02292: integrity constraint (BTO_YAMW.FK_RESTRICTION_04) violated - child record found

        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:494)
        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:446)
        at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1054)
        at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:623)
        at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:252)
        at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:612)
        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:226)
        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:59)
        at oracle.jdbc.driver.T4CPreparedStatement.executeForRows(T4CPreparedStatement.java:910)
        at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1119)
        at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3780)
        at oracle.jdbc.driver.T4CPreparedStatement.executeInternal(T4CPreparedStatement.java:1343)
        at oracle.jdbc.driver.OraclePreparedStatement.executeLargeUpdate(OraclePreparedStatement.java:3865)
        at oracle.jdbc.driver.OraclePreparedStatement.executeUpdate(OraclePreparedStatement.java:3845)
        at oracle.jdbc.driver.OraclePreparedStatementWrapper.executeUpdate(OraclePreparedStatementWrapper.java:1061)
        at org.jboss.jca.adapters.jdbc.WrappedPreparedStatement.executeUpdate(WrappedPreparedStatement.java:493)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:186)
        ... 131 more
Caused by: Error : 2292, Position : 0, Sql = delete from TAMW_resourceGroup where id=:1  and v=:2 , OriginalSql = delete from TAMW_resourceGroup where id=? and v=?, Error Msg = ORA-02292: integrity constraint (BTO_YAMW.FK_RESTRICTION_04) violated - child record found

        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:498)
        ... 147 more

2017-12-13 17:45:16,652 logLev=WARNING class=javax.enterprise.resource.webcontainer.jsf.flash thread="ajp-/0.0.0.0:8009-6" JSF1095: The response was already committed by the time we tried to set the outgoing cookie for the flash.  Any values stored to the flash will not be available on the next request.
```

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceBoundary.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceBoundary.java
@@ -306,11 +306,6 @@ public class ResourceBoundary {
         doRemoveResourceEntity(deletingOwner, resourceId);
     }
 
-    public void deleteApplicationById(ForeignableOwner deletingOwner, int applicationResId) throws ResourceNotFoundException,
-            ResourceNotDeletableException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
-        doRemoveResourceEntity(deletingOwner, applicationResId);
-    }
-
     private void doRemoveResourceEntity(ForeignableOwner deletingOwner, Integer resourceId) throws ResourceNotFoundException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
 
         ResourceEntity resourceEntity = commonService.getResourceEntityById(resourceId);
@@ -327,6 +322,7 @@ public class ResourceBoundary {
                 contextDomainService.getGlobalResourceContextEntity(), Action.DELETE, resourceEntity, resourceEntity.getResourceType())) {
             throw new NotAuthorizedException();
         }
+        permissionBoundary.removeAllRestrictionsForResourceGroup(resourceEntity.getResourceGroup());
 
         Integer groupId = resourceEntity.getResourceGroup().getId();
         //Special logic for the removal of an application server: If the current application server instance is the only release previously consuming this resource, the relation has to be attached to the default application server container.
@@ -345,6 +341,7 @@ public class ResourceBoundary {
             }
 
         }
+
         log.info("Resource with id: " + resourceEntity.getId() + " is going to be removed from the database...");
         resourceRepository.remove(resourceEntity);
 

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceRepository.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceRepository.java
@@ -21,6 +21,7 @@
 package ch.puzzle.itc.mobiliar.business.resourcegroup.control;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.logging.Logger;
 
@@ -32,6 +33,7 @@ import ch.puzzle.itc.mobiliar.business.deploy.entity.DeploymentEntity;
 import ch.puzzle.itc.mobiliar.business.releasing.entity.ReleaseEntity;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceEntity;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceGroupEntity;
+import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.ConsumedResourceRelationEntity;
 import ch.puzzle.itc.mobiliar.common.util.DefaultResourceTypeDefinition;
 
 public class ResourceRepository {

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceTypeDomainService.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceTypeDomainService.java
@@ -35,6 +35,7 @@ import javax.persistence.TypedQuery;
 import javax.persistence.criteria.*;
 
 import ch.puzzle.itc.mobiliar.business.security.control.PermissionService;
+import ch.puzzle.itc.mobiliar.business.security.control.RestrictionRepository;
 import ch.puzzle.itc.mobiliar.business.security.entity.Action;
 import org.apache.commons.lang.StringUtils;
 
@@ -75,6 +76,9 @@ public class ResourceTypeDomainService {
 
 	@Inject
 	private ResourcesScreenQueries queries;
+
+	@Inject
+	RestrictionRepository restrictionRepository;
 
 	public List<ResourceTypeEntity> getAllResourceTypesWithoutChildren() {
 		return entityManager.createQuery("select n from ResourceTypeEntity n order by n.name asc").getResultList();
@@ -172,6 +176,7 @@ public class ResourceTypeDomainService {
 	@HasPermission(permission = Permission.RESOURCETYPE, action = Action.DELETE)
 	public void removeResourceType(Integer resourceTypeId) throws ResourceNotFoundException, ResourceTypeNotFoundException {
 		ResourceTypeEntity resourceTypeEntity = commonService.getResourceTypeEntityById(resourceTypeId);
+		restrictionRepository.deleteAllWithResourceType(resourceTypeEntity);
 		removeResourceTypeEntity(resourceTypeEntity);
 	}
 

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java
@@ -715,4 +715,9 @@ public class PermissionBoundary implements Serializable {
         return permissionService.getCurrentUserName();
     }
 
+    @HasPermission(permission = Permission.RESOURCE, action = Action.DELETE, resourceSpecific = true)
+    public void removeAllRestrictionsForResourceGroup(ResourceGroupEntity resourceGroup) {
+        restrictionRepository.deleteAllWithResourceGroup(resourceGroup);
+    }
+
 }

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/control/RestrictionRepository.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/control/RestrictionRepository.java
@@ -21,6 +21,8 @@
 package ch.puzzle.itc.mobiliar.business.security.control;
 
 import ch.puzzle.itc.mobiliar.business.environment.entity.ContextEntity;
+import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceGroupEntity;
+import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceTypeEntity;
 import ch.puzzle.itc.mobiliar.business.security.entity.RestrictionEntity;
 import ch.puzzle.itc.mobiliar.business.utils.BaseRepository;
 
@@ -65,6 +67,26 @@ public class RestrictionRepository extends BaseRepository<RestrictionEntity> {
     public void deleteAllWithContext(ContextEntity context) {
         entityManager.createQuery("delete from RestrictionEntity r where r.context =:context")
                 .setParameter("context", context).executeUpdate();
+    }
+
+    /**
+     * Deletes all Restrictions matching a specific ResourceGroup
+     *
+     * @param resourceGroup ResourceGroupEntity to match
+     */
+    public void deleteAllWithResourceGroup(ResourceGroupEntity resourceGroup) {
+        entityManager.createQuery("delete from RestrictionEntity r where r.resourceGroup =:resourceGroup")
+                .setParameter("resourceGroup", resourceGroup).executeUpdate();
+    }
+
+    /**
+     * Deletes all Restrictions matching a specific ResourceType
+     *
+     * @param resourceType ResourceTypeEntity to match
+     */
+    public void deleteAllWithResourceType(ResourceTypeEntity resourceType) {
+        entityManager.createQuery("delete from RestrictionEntity r where r.resourceType =:resourceType")
+                .setParameter("resourceType", resourceType).executeUpdate();
     }
 
 }

--- a/AMW_business/src/test/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceBoundaryPersistenceTest.java
+++ b/AMW_business/src/test/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceBoundaryPersistenceTest.java
@@ -22,16 +22,24 @@ package ch.puzzle.itc.mobiliar.business.resourcegroup.boundary;
 
 import ch.puzzle.itc.mobiliar.business.domain.commons.CommonDomainService;
 import ch.puzzle.itc.mobiliar.business.environment.control.ContextDomainService;
+import ch.puzzle.itc.mobiliar.business.environment.entity.ContextEntity;
+import ch.puzzle.itc.mobiliar.business.foreignable.control.ForeignableService;
 import ch.puzzle.itc.mobiliar.business.foreignable.entity.ForeignableOwner;
+import ch.puzzle.itc.mobiliar.business.foreignable.entity.ForeignableOwnerViolationException;
 import ch.puzzle.itc.mobiliar.business.releasing.control.ReleaseMgmtPersistenceService;
 import ch.puzzle.itc.mobiliar.business.releasing.entity.ReleaseEntity;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceGroupRepository;
+import ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceRepository;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.control.ResourceTypeProvider;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.Resource;
+import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceEntity;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceGroupEntity;
 import ch.puzzle.itc.mobiliar.business.resourcegroup.entity.ResourceTypeEntity;
 import ch.puzzle.itc.mobiliar.business.security.boundary.PermissionBoundary;
+import ch.puzzle.itc.mobiliar.business.security.entity.Action;
+import ch.puzzle.itc.mobiliar.business.security.entity.Permission;
 import ch.puzzle.itc.mobiliar.common.exception.ElementAlreadyExistsException;
+import ch.puzzle.itc.mobiliar.common.exception.ResourceNotDeletableException;
 import ch.puzzle.itc.mobiliar.common.exception.ResourceNotFoundException;
 import ch.puzzle.itc.mobiliar.common.exception.ResourceTypeNotFoundException;
 import ch.puzzle.itc.mobiliar.test.testrunner.PersistenceTestRunner;
@@ -51,6 +59,8 @@ import java.util.logging.Logger;
 
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @RunWith(PersistenceTestRunner.class)
@@ -68,6 +78,12 @@ public class ResourceBoundaryPersistenceTest {
 
     @Mock
     private ResourceTypeProvider resourceTypeProvider;
+
+    @Mock
+    ForeignableService foreignableService;
+
+    @Mock
+    ResourceRepository resourceRepository;
 
     @Mock
     ResourceGroupRepository resourceGroupRepository;
@@ -161,4 +177,36 @@ public class ResourceBoundaryPersistenceTest {
         // when // then
         resourceBoundary.createNewResourceByName(ForeignableOwner.getSystemOwner(), "test", asType.getId(), release1.getId());
     }
+
+    @Test
+    public void shouldDeleteRestrictionsWhenDeletingResource() throws ElementAlreadyExistsException, ResourceTypeNotFoundException,
+            ResourceNotFoundException, ResourceNotDeletableException, ForeignableOwnerViolationException {
+        // given
+        ResourceTypeEntity appType = new ResourceTypeEntity();
+        appType.setName("application");
+        entityManager.persist(appType);
+        ReleaseEntity release1 = new ReleaseEntity();
+        release1.setName("release1");
+        entityManager.persist(release1);
+
+        when(releaseService.getById(release1.getId())).thenReturn(release1);
+        when(commonService.getResourceTypeEntityById(appType.getId())).thenReturn(appType);
+        when(permissionBoundary.canCreateResourceInstance(appType)).thenReturn(Boolean.TRUE);
+        when(resourceGroupRepository.loadUniqueGroupByNameAndType("app1", appType.getId())).thenReturn(null);
+        when(resourceGroupRepository.getResourceGroupByName("app1")).thenReturn(null);
+        Resource r1 = resourceBoundary.createNewResourceByName(ForeignableOwner.getSystemOwner(), "app1", appType.getId(),
+                release1.getId());
+        when(commonService.getResourceEntityById(r1.getId())).thenReturn(r1.getEntity());
+        when(permissionBoundary.hasPermission(any(Permission.class), any(ContextEntity.class), any(Action.class),
+                any(ResourceEntity.class), any(ResourceTypeEntity.class))).thenReturn(true);
+        when(resourceGroupRepository.find(r1.getEntity().getResourceGroup().getId())).thenReturn(r1.getEntity().getResourceGroup());
+
+        // when
+        resourceBoundary.removeResource(ForeignableOwner.AMW, r1.getId());
+
+        // then
+        verify(permissionBoundary).removeAllRestrictionsForResourceGroup(r1.getEntity().getResourceGroup());
+        verify(resourceGroupRepository).remove(r1.getEntity().getResourceGroup());
+    }
+
 }

--- a/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/applist/ApplistView.java
+++ b/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/applist/ApplistView.java
@@ -103,11 +103,8 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
     @Inject
     private ForeignableBoundary foreignableBoundary;
 
-
 	@PostConstruct
 	public void init() {
-
-
 		if (filterReleaseSelector == null) {
 			filterReleaseSelector = new ReleaseSelector(null, releaseDataProvider.getReleaseMap());
 		}
@@ -132,7 +129,6 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 						.getUpcomingReleaseId());
 			}
 		}
-
 		loadAppServerList();
 	}
 
@@ -200,8 +196,6 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 		}
 		return NavigationUtils.getRefreshOutcomeWithAdditionalParam(query.toString());
 	}
-
-
 	public void setRelease(Integer releaseId) {
 		this.filterReleaseSelector.setSelectedReleaseId(releaseId);
 	}
@@ -210,18 +204,15 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 		appServerName = null;
 	}
 
-
 	public Integer getMaxResults() {
 		return maxResults == null ? 0 : maxResults;
 	}
-
 
 	@Override
 	public void afterAddingAppOrAs() {
 		loadAppServerList();
 		clearPopupFields();
 	}
-
 
 	/**
 	 * Applikation entfernen
@@ -238,8 +229,8 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 					String message = "Application successfully deleted";
 					GlobalMessageAppender.addSuccessMessage(message);
 					isSuccessful = true;
-				}catch(EJBException e){
-					if(e.getCause() instanceof NotAuthorizedException) {
+				} catch(EJBException e) {
+					if (e.getCause() instanceof NotAuthorizedException) {
 						GlobalMessageAppender.addErrorMessage(e.getCause().getMessage());
 					} else {
 						throw e;
@@ -274,8 +265,8 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 					String message = "Applicationserver successfully deleted";
 					GlobalMessageAppender.addSuccessMessage(message);
 					isSuccessful = true;
-				}catch(EJBException e){
-					if(e.getCause() instanceof NotAuthorizedException) {
+				} catch(EJBException e) {
+					if (e.getCause() instanceof NotAuthorizedException) {
 						GlobalMessageAppender.addErrorMessage(e.getCause().getMessage());
 					} else {
 						throw e;
@@ -288,13 +279,12 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 		} catch (ResourceNotFoundException e) {
 			String errorMessage = "Could not load selected server for deletion.";
 			GlobalMessageAppender.addErrorMessage(errorMessage);
-		} catch(Exception e){
+		} catch (Exception e) {
 			String errorMessage = "Could not delete selected application server.";
 			GlobalMessageAppender.addErrorMessage(errorMessage);
 		}
 		return isSuccessful;
 	}
-
 
 	private List<ResourceWithRelations> loadAppServers(String filter, Integer maxResults, ReleaseEntity release) {
 		if(maxResults!=null && maxResults==0) {
@@ -304,28 +294,24 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 	}
 
 	public boolean canCreateApplicationServerInstance(){
-
 		return permissionBoundary.canCreateResourceInstance(DefaultResourceTypeDefinition.APPLICATIONSERVER);
 	}
+
 	public boolean canCreateApplicationInstance(){
 		return permissionBoundary.canCreateResourceInstance(DefaultResourceTypeDefinition.APPLICATION);
 	}
 
     public boolean canShowDeleteApp(ResourceEntity app){
-		// TODO add context check
         return permissionBoundary.hasPermission(Permission.RESOURCE, null, Action.DELETE, app, null) && foreignableBoundary.isModifiableByOwner(ForeignableOwner.getSystemOwner(), app);
     }
 
     public boolean canShowDeleteAppServer(ResourceWithRelations appServer){
         ResourceEntity appserverResource = appServer.getResource();
-
-		// TODO add context check
         return appserverResource.isDeletable() && permissionBoundary.hasPermission(Permission.RESOURCE, null, Action.DELETE, appserverResource, null) && foreignableBoundary.isModifiableByOwner(ForeignableOwner.getSystemOwner(), appserverResource);
     }
 
     public ForeignableAttributesDTO getForeignableAttributes(ResourceEntity app){
         return new ForeignableAttributesDTO(app.getOwner(), app.getExternalKey(), app.getExternalLink());
     }
-
 
 }

--- a/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/applist/ApplistView.java
+++ b/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/applist/ApplistView.java
@@ -234,7 +234,7 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 				GlobalMessageAppender.addErrorMessage(errorMessage);
 			} else {
 				try{
-					resourceBoundary.deleteApplicationById(ForeignableOwner.getSystemOwner(), applicationId);
+					resourceBoundary.removeResource(ForeignableOwner.getSystemOwner(), applicationId);
 					String message = "Application successfully deleted";
 					GlobalMessageAppender.addSuccessMessage(message);
 					isSuccessful = true;
@@ -249,12 +249,9 @@ public class ApplistView implements Serializable, ApplicationCreatorDataProvider
 		} catch (ResourceNotFoundException e) {
 			String errorMessage = "Could not load selected application for deletation.";
 			GlobalMessageAppender.addErrorMessage(errorMessage);
-		} catch (ResourceNotDeletableException e) {
-            String errorMessage = "The selected application can not be deleted.";
-            GlobalMessageAppender.addErrorMessage(errorMessage);
-        }catch (ForeignableOwnerViolationException e){
+		} catch (ForeignableOwnerViolationException e){
             GlobalMessageAppender.addErrorMessage("Application with id "+applicationId+" can not be deleted by owner "+e.getViolatingOwner());
-        }catch (Exception e) {
+        } catch (Exception e) {
 			String errorMessage = "Could not delete selected application.";
 			GlobalMessageAppender.addErrorMessage(errorMessage);
 		}

